src/api/wasm.c

```diff
@@ -419,7 +419,14 @@ m3ApiRawFunction(wasmtic_fget)
     
     tic_mem* tic = (tic_mem*)getWasmCore(runtime);
 
-    m3ApiReturn(tic_api_fget(tic, sprite_index, flag));
+    if (flag == -1)
```

Comment:
Minimally `m3ApiReturnType` would need to change to `uint8_t` and we'd need to make sure that doesn't break the binary API of any of the existing target WASM languages.  If they typecast 0 to falsy and non-zero to truthy we might be ok here.

Reply:
hey @joshgoebel, good catch, I missed the return type statement. Could I do something like 
```
if (flag == -1) {
    m3ApiReturnType(int);
    m3ApiReturn(tic_api_fget_all(tic, sprite_index));
    } else {
        m3ApiReturnType(bool);
        m3ApiReturn(tic_api_fget(tic, sprite_index, flag));
    }
```
maybe? What do you think?

Reply:
No, the WASM<=>outside API must be static.  The params (type and number) must always be the same, the return type must always be the same.  But when I say "binary compatibility" I'm referring to what any cartridges are already expecting that have been compiled.

All these params are really u32 or u64 going across the wire (the only true WASM primitives)... but how they are interpreted on the WASM VM side is what matters.  You can't change that by changing TIC-80...

That's the only reason I suggest we might be able to get away with changing `bool` to `uint_8t` here... it all depends on how the various compilers are typecasting those values.

Reply:
ok yes, I see what you mean. How would we proceed about this then? It could also become a new api function, like `fgetall` or something like that. Like I mentioned, since I don't know how to test wasm, and apparently, it seems kindof impossible to test if there are an unknown amount of compilers that compiles to wasm, we're a bit in a pickle in terms of quality assurance...

Reply:
Only the ones we have libs for would need to be tested.  That's only like 2-3?

Reply:
ok i'll update the pull request then. Do you know how to test them?

Reply:
Sure.  Write a simple program in each language that calls the API expecting it to return a bool - printing both cases to the screen, etc. Compile and test.

Then run that same binary cartridge on a new compiled version of TIC-80 that's changed the binary API, and see if it still works as expected.  The wiki has instructions for working with WASM cartridges.